libs/algo/odc/algo/_masking.py

```diff
@@ -383,32 +383,32 @@ def binary_closing(xx: xr.DataArray, radius: int = 1, **kw) -> xr.DataArray:
     return xr_apply_morph_op(xx, "closing", radius, **kw)
 
 
-def mask_cleanup_np(mask: np.ndarray, filter: Dict[str, int] = dict(closing=0, opening=2, dilation=5)) -> np.ndarray:
+def mask_cleanup_np(
+    mask: np.ndarray,
+    mask_filters: Iterable[Tuple[str, int]] = [("opening", 2), ("dilation", 5)],
+) -> np.ndarray:
     """
-    Apply morphological closing(>0) then opening(>0) followed by dilation(>0) on given binary mask.
+    Apply morphological operations on given binary mask.
 
     :param mask: Binary image to process
-    :param filter: dict of integer (closing=int, opening=int, dilation=int), order of operation
-        closing = remove small holes in cloud - morphological closing
-        opening = shrinks away small areas of the mask
-        dilation = adds padding to the mask
+    :param mask_filters: Iterable tuples of morphological operations to apply on mask
     """
     import skimage.morphology as morph
 
     assert mask.dtype == "bool"
 
-    if filter.get("closing", 0) == 0 and filter.get("opening", 0) == 0 and filter.get("dilation", 0) == 0:
-        return mask
+    for operation, radius in mask_filters:
+        if operation == "closing" and radius > 0:
+            mask = morph.binary_closing(mask, _disk(radius, mask.ndim))
 
-    if filter.get("closing", 0) > 0:
-        mask = morph.binary_closing(mask, _disk(filter["closing"], mask.ndim))
+        if operation == "opening" and radius > 0:
+            mask = morph.binary_opening(mask, _disk(radius, mask.ndim))
 
-    if filter.get("opening", 0) > 0:
-        mask = morph.binary_opening(mask, _disk(filter["opening"], mask.ndim))
-
-    if filter.get("dilation", 0) > 0:
-        mask = morph.binary_dilation(mask, _disk(filter["dilation"], mask.ndim))
+        if operation == "dilation" and radius > 0:
+            mask = morph.binary_dilation(mask, _disk(radius, mask.ndim))
 
+        if operation == "erosion" and radius > 0:
+            mask = morph.binary_erosion(mask, _disk(radius, mask.ndim))
```

Comment:
Code above silently ignore typos in the operation string.

A common pattern to avoid duplication like that is to use a dictionary that maps operation name to a function to call. Like below:

```python
ops = dict(
opening=morph.binary_opening, 
closing=morph.binary_closing,
 ...)
for operation, radius in mask_filters:
     op = ops.get(operation, None)
     if op is None:
         raise ValueError(f"Not supported mophological operation: {operation}")
     if radius > 0:
         mask = op(mask, _disk(radius, mask.ndim))
```